src/freertos_drivers/ti/TivaCan.cxx

```diff
@@ -155,13 +177,20 @@ void TivaCan::interrupt_handler()
         {
             /* bus off error condition */
             ++busOffCount;
+            canState = CAN_STATE_BUS_OFF;
+
+            /* flush data in the tx pipeline */
+            txBuf->flush();
+            txPending = false;
+            txBuf->signal_condition_from_isr();
         }
         if (status & CAN_STATUS_EWARN)
         {
             /* One of the error counters has exceded a value of 96 */
             ++softErrorCount;
-            /* flush and data in the tx pipeline */
-            MAP_CANMessageClear(base, 2);
```

Comment:
Are you sure you want to remove this line of code? You could end up with a stale message in the CAN object when the bus recovers.

Reply:
Yes, I want to keep a message in the transmit object, so that the CAN controller keeps attempting to transmit something periodically. Otherwise we might never realize the cable was plugged in because the bus is quiescent.
I made another change that causes the CAN frame be replaced with newly transmitted frames every time ::write() is called. This way the frame in the transmit object is not going to be super old, provided that there are regular transmissions happening.